MidiTools/src/main/kotlin/com/example/android/miditools/MidiCiInitiator.kt

```diff
@@ -134,6 +241,15 @@ class MidiCiInitiator {
         return (message[0] == targetFirstByte.toByte())
     }
 
+    private fun verifyMessageIsGrouplessMessage(message: ByteArray): Boolean {
+        if ((message.isEmpty()) || (message.size % UMP_PACKET_MULTIPLE != 0)) {
+            return false
+        }
+
+        val targetFirstByte = (GROUPLESS_MESSAGE_TYPE shl 4) // mt = f, f = 0
```

Comment:
This byte contains the top two bits of the status field.
So you should mask off the bottom bits of message[0]